sql/catalyst/src/main/antlr4/org/apache/spark/sql/catalyst/parser/SqlBaseParser.g4

```diff
@@ -434,17 +434,31 @@ resource
 dmlStatementNoWith
     : insertInto query                                                             #singleInsertQuery
     | fromClause multiInsertQueryBody+                                             #multiInsertQuery
-    | DELETE FROM multipartIdentifier tableAlias whereClause?                      #deleteFromTable
-    | UPDATE multipartIdentifier tableAlias setClause whereClause?                 #updateTable
-    | MERGE INTO target=multipartIdentifier targetAlias=tableAlias
-        USING (source=multipartIdentifier |
+    | DELETE FROM tableReference tableAlias whereClause?                           #deleteFromTable
+    | UPDATE tableReference tableAlias setClause whereClause?                      #updateTable
+    | MERGE INTO target=tableReference targetAlias=tableAlias
+        USING (source=tableReference |
           LEFT_PAREN sourceQuery=query RIGHT_PAREN) sourceAlias=tableAlias
         ON mergeCondition=booleanExpression
         matchedClause*
         notMatchedClause*
         notMatchedBySourceClause*                                                  #mergeIntoTable
     ;
 
+tableReference
+    : IDENTIFIER_KW LEFT_PAREN expression RIGHT_PAREN
+    | multipartIdentifier
+    ;
+
+identifierReference
+    : IDENTIFIER_KW LEFT_PAREN expression RIGHT_PAREN
+    | multipartIdentifier
+    ;
+identifierReferenceAllowTemp
```

Comment:
This does not seem any different than `identifierReference` above, or `tableReference`?

Reply:
+1 with @dtenedor . Shall we have just one identifierReference for all the multipartIdentifier

Reply:
That doesn't work, the context in which they are parsed influences the resolution. E.g. whether they produce a UnresolvedFunctionName, or an UnresolvedRelation, etc....

Reply:
Or we can ask the caller side (e.g. the parser of CREATE TABLE command) to create the proper identifier logical plan, but I think using antlr rules to track the context is simpler.

BTW I think we can still have a `multipartIdentifierOrClause` to avoid duplicated code.

Reply:
@cloud-fan Can you sketch out what that would look like? What would its visit return a tuple of (Seq[String], expression) ?